### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by sainingo**
CreateD model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
Set up associations between models.
###  Models should include some additional methods.

**Users**
- A method that returns the 3 most recent posts for a given user.
**Posts**
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
**Comments**
- A method that updates the comments counter for a post.
**Likes**
- A method that updates the likes counter for a post.
- Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_many :posts, class_name: 'post', foreign_key: 'author_id'
+  has_many :comments, class_name: 'comment', foreign_key: 'author_id'
+  has_many :likes, class_name: 'like', foreign_key: 'author_id'
 end
```
Comment (thecodechaser) [inline app/models/user.rb]: - [ ] Please add the following methods to the `User` model.

image

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
 end
```
Comment (thecodechaser) [inline app/models/post.rb]: - [ ] Please add the following methods in the `Post` model.

image

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
 class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
 end
```
Comment (thecodechaser) [inline app/models/comment.rb]: - [ ] Please add the following methods in the `Comment` model.

image

File: app/models/like.rb [hunk 4]
```diff
@@ -1,2 +1,6 @@
 class Like < ApplicationRecord
+  # belongs_to :user
+  # belongs_to :post
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
 end
```
Comment (thecodechaser) [inline app/models/like.rb]: - [ ] Please add the following methods in the `Like` model.

image

**Review by thecodechaser — APPROVED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @sainingo👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elyor-doniyorov — APPROVED** [review]
Hi @sainingo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Implemented requested changes

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**